### PR TITLE
[Spark] Uses java-based coordinated commits classes in Delta Spark

### PR DESCRIPTION
**Pull request by LukasRupprecht**
#### Which Delta project/connector is this regarding?

- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description

This PR adopts the coordinated commits interfaces from the storage module in Delta Spark. It removes the existing scala classes and adds the necessary conversion code from java -> scala (and in the opposite direction) where necessary.

## How was this patch tested?

Adds some unit tests for the critical code pieces (action serialization/deserialization and LogStore conversion). For the remainder, existing tests are sufficient.

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

File: kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/CommitCoordinatorProviderSuite.scala [hunk 1]
```diff
@@ -159,7 +159,7 @@ protected trait TestCommitCoordinatorClientBase extends CommitCoordinatorClient
     version: Long,
     lastKnownBackfilledVersion: lang.Long): Unit = {}
 
-  override def semanticEquals(other: CommitCoordinatorClient): lang.Boolean = this == other
+  override def semanticEquals(other: CommitCoordinatorClient): Boolean = this == other
```
Comment (prakharjain09) [inline kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/CommitCoordinatorProviderSuite.scala:162]: [nit] JBoolean

File: kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/CommitCoordinatorProviderSuite.scala [hunk 2]
```diff
@@ -200,7 +200,7 @@ class Builder2(hadoopConf: Configuration) extends CommitCoordinatorBuilder(hadoo
 // Test 3
 // Commit Coordinator Client with semanticEquals implemented for testing
 class TestCommitCoordinatorClient3(val key: String) extends TestCommitCoordinatorClientBase {
-  override def semanticEquals(other: CommitCoordinatorClient): lang.Boolean =
+  override def semanticEquals(other: CommitCoordinatorClient): Boolean =
```
Comment (prakharjain09) [inline kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/CommitCoordinatorProviderSuite.scala:203]: [nit] JBoolean

**Review by prakharjain09 — APPROVED** [review]

File: kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/CommitCoordinatorProviderSuite.scala (hunk 2 above)
Comment (LukasRupprecht) [reply kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/CommitCoordinatorProviderSuite.scala:203]: I will fix all the imports in a follow up.

File: kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/CommitCoordinatorProviderSuite.scala (hunk 1 above)
Comment (prakharjain09) [reply kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/CommitCoordinatorProviderSuite.scala:162]: This PR is already very big. Maybe we can avoid using java classes directly and rename them to JBoolean, JMap, JLogStore etc in a followup PR.

**Pull request merged**